examples/User Deprovisioning/Users_Deprovision.ps1

```diff
@@ -179,51 +193,80 @@ Function Start-Deprovisioning-Script {
-    try {
-        $RootFolderResp = "$(box folders:items 0 --sort=name --direction=ASC --json 2>&1)"
-        $RootFolder = $RootFolderResp | ConvertFrom-Json
-    } catch {
-        Write-Log "Could not get root directory for current user (ID: $UserId). See log for details. " -errorMessage $RootFolderResp -output true -color Red
-        break
-    }
-
-    # Check if "Employee Archive" folder already exists
-    ForEach($FolderItem in $RootFolder) {
-        if($FolderItem.name -eq $EmployeeArchiveFolderName) {
-            $EmployeeArchiveFolderID = $FolderItem.id
-            Write-Log "'$EmployeeArchiveFolderName' folder already exists with folder ID: $($EmployeeArchiveFolderID)." -output true
-            break
+    # Create folder if need to transfer content
+    if ($TransferContent -eq "Y") {
+        # Check if new file owner ID already specified
+        if ($NewFilesOwnerID) {
+            $UserId = $NewFilesOwnerID
+        } elseif (!(Assert-IsNonInteractiveShell)) {
+            Write-Log "Please specify the user ID of the user who will own the files of the users being deprovisioned." -output true -color Yellow
+            Write-Log "Press Enter if you want to use the current user as the new owner." -output true -color Yellow
+            $UserId = Read-Host "User ID"
         }
-    }
-
-    # Create new "Employee Archive" folder if it doens't exist
-    if($null -eq $EmployeeArchiveFolderID) {
-        if(!$DryRun) {
+        
+        # No user ID specified or in non-interactive mode
+        if (!$UserId) {
+            Write-Log "No user ID specified. Using current user as the new files owner." -output true -color Yellow
             try {
-                $EmployeeArchiveFolderResp = "$(box folders:create 0 "$EmployeeArchiveFolderName" --fields="id" --json 2>&1)"
-                $EmployeeArchiveFolderID = $EmployeeArchiveFolderResp | ConvertFrom-Json | ForEach-Object { $_.id }
-                Write-Log "Successfully created new '$EmployeeArchiveFolderName' root folder with ID: $($EmployeeArchiveFolderID)." -output true
-                Write-Log $EmployeeArchiveFolderResp
+                $UserResp = "$(box users:get --json 2>&1)"
+                $User = $UserResp | ConvertFrom-Json
+                $UserId = $User.id 
+                Write-Log "Successfully get current user: $($User.login), ID: $($User.id)." -output true
+                Write-Log $UserResp
             } catch {
-                Write-Log "Could not create new '$EmployeeArchiveFolderName' root folder. See log for details." -errorMessage $EmployeeArchiveFolderResp -output true -color Red
+                Write-Log "Could not get the current user. See log for details." -errorMessage $UserResp -output true -color Red
+                break
+            }
+        }
+        
+        # Check if user ID is valid
+        if ($UserId) {
+            try {
+                $UserResp = "$(box users:get --as-user=$UserId --json 2>&1)"
```

Comment:
When running it as current user (not providing `NewFilesOwnerID`), I get error: 
> Could not get the user with ID 18057416283. See log for details.
> Unexpected API Response [403 Forbidden | dbqcbfh68lep41n2.0d6fa578496c7f1e78aacf6a4a76b92ab] access_denied_insufficient_permissions - Access denied - insufficient permission

I tried calling the raw command:
`box users:get --as-user=18057416283` (18057416283 is my current user id) and get above error, but running `box users:get ` without any issue

Reply:
Maybe your scope is missing the option "Make API calls using the as-user header" so it's not allowing you to run command as another user.

Reply:
No, it works when I provide another user ID and everything passes. But when i don't specify `NewFilesOwnerID`, so it uses current user id.